nexus/lib/qmcpack.py

```diff
@@ -525,6 +547,38 @@ def write_prep(self):
                         fobj.close()
                     #end if
                 #end for
+                grand_canonical_twist_average = 'nelecs_at_twist' in self
+                if grand_canonical_twist_average:
+                    for itwist, qi in enumerate(input.inputs):
+                        elecs = self.nelecs_at_twist[itwist]
+                        # step 1: resize particlesets
+                        nup = elecs[0]
+                        ndn = elecs[1]
+                        qi.get('u').set(size=nup)
+                        qi.get('d').set(size=ndn)
+                        # step 2: resize orbital sets
+                        sb = qi.get('sposet_builder')
+                        bb = sb.bspline  # hard-code for Bspline orbs
+                        assert itwist == bb.twistnum
+                        sposets = bb.sposets
+                        if len(sposets) == 1:  # RHF/ROHF
```

Comment:
This won't work if there are other sposets in the builder (e.g. density matrices).  You should iterate over the determinants, and then pull out and resize only the referenced sposets from the builder.

Reply:
yes, resizing determinant first, then resize only SPOs used is the better way to go
will fix in next push